mpmath/libmp/libmpf.py

```diff
@@ -33,7 +33,8 @@ class ComplexResult(ValueError):
     pass
 
 # All supported rounding modes
-round_nearest = sys.intern('n')
+round_nearest = sys.intern('n')  # Round to nearest with tying away from zero
+round_nearest_even = sys.intern('m')  # Round to nearest with tying to even
```

Comment:
No, it's wrong.  Currently, round_nearest is exactly "round to nearest with ties to even" (like roundTiesToEven from the IEEE 754 standard).  See e.g. round_int().

It seems, you don't add any code, that supports new meaning (outside of to_str()).  Anyway, this is a backward compatibility break, which I won't introduce.

Reply:
> No, it's wrong. Currently, round_nearest is exactly "round to nearest with ties to even" (like roundTiesToEven from the IEEE 754 standard). See e.g. round_int().

I incorrectly assumed that the default in mpmath was tie away from zero because that's what `to_str` does:
```python
>>> nstr(mp.mpf('0.25'), 1)
'0.3'
>>> nstr(mp.mpf('0.0625'), 2)
'0.063'
```

> It seems, you don't add any code, that supports new meaning (outside of to_str()). Anyway, this is a backward compatibility break, which I won't introduce.

I did, please see lines 1128 -- 1141 of libmpf.py (also the new tests in test_format.py).

Reply:
> I incorrectly assumed that the default in mpmath was tie away from zero because that's what to_str does

Mpmath uses binary arithmetic.  Here defaults for the mp context:
```pycon
>>> from mpmath import mp
>>> mp._prec_rounding
[53, 'n']
```

to_str() for decimal output, probably, has historically different defaults.

> I did, please see lines 1128 -- 1141

round_digits() used only for formatting.